File: codes/java/src/hello/Entry.java
package hello;

public class Entry {
    public static void main(String[] args) {

        if (args.length < 1) {
            System.out.println("Please enter an integer value");
            System.exit(0);
        }

        int n = Integer.parseInt(args[0]);
        System.out.println("Fact(" + n + ")= " + fact(n));

    }

    public static int fact(int i) {
        if (i <=1)	return 1;
        return i * fact(i-1);
    }

}

File: codes/lua/hello/entry.lua
fact = function (i)
    if i <= 0 then return 1 end
    return i * fact(i-1)
end

if #arg < 1 then
    print ("Please enter an integer value")
    os.exit()
end

n = tonumber(arg[1])
print ("Fact(" .. n .. ")= " .. fact(n))
